README.md

```diff
@@ -177,44 +177,65 @@ Example of a partial CR:
 
 ### Mounting and injecting binding information
 
-This specification allows for data to be mounted using volumes or injected using environment variables.  The best practice is to mount any sensitive information, such as passwords, since that will avoid accidentally exposure via environment dumps and subprocesses.  Also, binding binary data (e.g. .p12 certificate for Kafka) as an environment variable might cause a pod to fail to start (stuck on `CrashLoopBackOff`), so it advisable for backing services with such binding data to mark it with `bindAs: volume`.
+#### Mounting data
+Implementations of this specification **MUST** mount the binding data into the consuming application container in the following location:
+
+```
+<mountPathPrefix>/<service-id>/secret/<persisted_secret>
+```
+
+Where:
+* `<mountPathPrefix>` defaults to `/platform/bindings` if not specified in the `ServiceBinding` CR via the `mountPathPrefix` element.  The environment variable `SERVICE_BINDINGS_ROOT` **MUST** be set to the value of `<mountPathPrefix>`, so that applications can always find this directory, and it is immutable.  This means that if another `ServiceBinding` CR wants to project itself into the same container, it **MUST** reuse the current `SERVICE_BINDINGS_ROOT` value.  
+* `<service-id>` is the `id` field of the corresponding `service` entry in the `ServiceBinding` CR.  If the `id` field is not present, the `name` field is used instead.  The `<service-id>` path **MUST** be unique between the services bound to a particular application.
+* `<persisted_secret>` represents a set of files where the filename is a Secret key and the file contents is the corresponding value of that key.
+
+Example:  `/platform/bindings/mongo-db/secret/MONGODB_HOST`
+
+#### Cross services data mappings
+
+The `ServiceBinding` CR allows for the declaration of mappings that are potentially composed of different service bindings (ids).  For this case, the special id of `dataMappings` is reserved and **MUST NOT** be used by any service bindings.  
```

Comment:
`dataMappings` vs `data-mappings`?
I couldn't find an official Linux naming convention guideline. But overall it seems like using lower-case/dashe is more common. e.g. https://developers.google.com/style/filenames

Reply:
Same goes for `envVars`

Reply:
that's a good question.  I guess for filenames that's more common, but should we view these as traditional files or more like k8s bindings?  My view is that yes they are files, but probably not in the sense of regular r/w OS files. 

In the binding schema we have `basePath` and `bindingRole`, which would become "files" in the persisted model.

Reply:
this is a good conversation, it looks like its more common in Linux file systems to have files with `-` to separate logical words. 

Cloud native buildpack specification also uses `-` as the separator here in the example https://github.com/buildpacks/spec/blob/master/extensions/bindings.md#example-directory-structure-1 and here it talks about the naming convention https://github.com/buildpacks/spec/blob/master/extensions/bindings.md#launch-phase

I think we should probably have a section that talks about naming convention, else we might have files that are in different naming formats. The spec also allows extra binding properties, if either the backing operator service or the custom name in the dataMappings portion chooses to use `-` in the binding name, then we might have few files which are using camel case and few with `-`. I think we should have consistent naming convention.

Reply:
I personally prefer `dash-delimited` over `camelCase`. It looks more familiar in Kubernetes/container.
e.g. [K8s resource naming convention](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names):
```
a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')" for field "metadata.name".
```

And the secret bindings generated by SBO will be following the k8s naming rules.
Having said that I'm good either way 🙂 

+1 for a naming convention (eventually).

Reply:
good point guys.  I will update the PR, and will open a separate issue to go through the whole spec and apply the name convention change (plus a notice about it).